File: java/src/main/java/mx/iteso/Publish.java
package mx.iteso;

/**
 * Class for Publish permissions.
 */
public class Publish extends Privilege {
    /**
     * adds user to test.
     */
    private User user;

    /**
     *
     * @param us adds the user using the permissions.
     */
    public Publish(final User us) {
        this.user = us;
    }

    /**
     *
     * @return user description.
     */
    @Override
    public String getDescripcion() {
        return user.getDescripcion() + "can publish";
    }

    /**
     *
     * @return publish permissions.
     */
    @Override
    public boolean canPublish() {
        return true;
    }

    /**
     *
     * @return comment permissions.
     */
    @Override
    public boolean canComment() {
        return user.canComment();
    }

    /**
     *
     * @return delete permissions.
     */
    @Override
    public boolean canDelete() {
        return user.canDelete();
    }

    /**
     *
     * @return approve permissions.
     */
    @Override
    public boolean canApprove() {
        return user.canApprove();
    }
}

File: python/publish.py
from PrivilegeDecorator import PrivilegeDecorator


class Publish(PrivilegeDecorator):
    PRIVILEGE = 'can publish'

    def __init__(self, newUser):
        self.user = newUser

    def getUser(self):
        return self.user

    def delete(self):
        return self.user.delete()

    def approve(self):
        return self.user.approve()

    def publish(self):
        return self.PRIVILEGE

    def comment(self):
        return self.user.comment()
